docs/specs.md

```diff
@@ -87,181 +113,323 @@ can be considered as _a collection of text chunks_. How those chunks are
-chunk, even when in streaming mode.
+#### Structural iteration
 
+* A document is considered as a sequence of _rows_
+* Each row contains a list of column values, each column value is a string with
+  the contents of the corresponding table cell
+* Iteration then produces a sequence of chunks, one per document **row**.
+* Each chunk contains the three usual fields: `id`, `data` and `context`. 
+* `id` and `context`have the same structure and semantics as for the other
+  document types
+* However, `data` is different: for table documents it contains not a text
+  field, but a _list_ of text fields, one per column in the row. Hence the
+  `data` field can also be iterated upon.
 
 
-#### Sequential source document
+#### Full iteration
 
-This is a simpler document model in which the document is divided into independent chunks, each one with three elements:
+For Table source documents in full iteration mode:
+ * A document is considered as a sequence of _cells_
+ * Cells are traversed in row-major order, top-to-bottom and left-to-right
+ * Traversing produces a linear list of document chunks
+ * A chunk contents has the same elements as in sequence documents: `id`,
+   `data` and `context`
+ * The `data` element contains the contents of a single document cell
+ * The context elements have the usual _global_, _before_ and _after_ fields
+   **plus** one additional field:
+     - _column_: contains the column name and index for the cell in this chunk
+ * Note that here _before_ and _after_ refer to the previous & next cell,
+   which for the first/last cells in a row will be in another row
 
-* **id**: an arbitrary string that should be unique per document. Its mission is to make it easier later on to map detected PII instances to the chunk they are part of
-* **text**: a text section that contains the textual contents of the chunk. It will be a string containing UTF-8 raw text. It can contain newlines or blank lines, to be considered as part of the text structure, but no formatting or layout contents (it is assumed that exact formatting & layout is lost when creating the source document for PII processing – this is a price we pay for simplicity)
-* **context**: (optional) additional field contributing context for the chunk.
-It can contain the same three context fields as with tabular documents:
-_global_ (context defined in the document header), _before_ (previous chunk)
-and _after_ (the next chunk).
+
+## Storage format
+
+As a support file format, we define YAML files as the serialized representation
+of a Source Document. Those are easy to inspect visually and handle/edit
+manually, and also can be processed via automatic tools and packages. The YAML
+specification is actually somehow complex, but we would use only a subset of it:
+the part strictly needed to support the definitions above
+ * sequences, for the sequences of chunks
+ * mappings, for each chunk
+ * literal block scalars, to hold the text contents of each chunk
+
```

Comment:
It should be noted that the actual location of the serialization of the YAML data doesn't have to be a "file". It could be a database, or any datastore for storing the data securely and for later retrieval.